src/components/Overlay/provider.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+import { RouteComponentProps, withRouter } from "react-router";
+import {
+  InnerOverlayContextInterface,
+  OverlayContext,
+  OverlayContextInterface,
+  OverlayTheme,
+  OverlayType
+} from "./context";
+
+class Provider extends React.Component<
+  RouteComponentProps<{}>,
+  OverlayContextInterface
+> {
+  notificationCloseDelay = 2500;
+  constructor(props) {
+    super(props);
+    this.state = {
+      context: null,
+      hide: this.hide,
+      show: this.show,
+      theme: null,
+      type: null
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.location.pathname !== prevProps.location.pathname) {
+      this.hide();
+    }
+  }
+
+  show = (
+    type: OverlayType,
+    theme?: OverlayTheme,
+    context?: InnerOverlayContextInterface
+  ) => {
+    this.setState({ type, theme, context });
+    document.body.style.overflow = type !== OverlayType.message ? "hidden" : "";
```

Comment:
![2r9nfi](https://user-images.githubusercontent.com/6833443/51252105-3ae19b80-199b-11e9-982e-727670a43686.jpg)

Reply:
Ask the author :P

Reply:
Oh okay. I'll open ticket for this.